pkg/helpers/apimachinery_status_conditions.go

```diff
@@ -0,0 +1,29 @@
+package helpers
+
+import (
+	"encoding/json"
+	"sort"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CopyConditions copies the set of conditions
+func CopyConditions(conditions []metav1.Condition) []metav1.Condition {
+	newConditions := make([]metav1.Condition, 0)
+	for idx := range conditions {
+		// copy
+		newCondition := conditions[idx]
+		newConditions = append(newConditions, newCondition)
+	}
+	return newConditions
+}
+
+// ConditionMarshal marshals the set of conditions as a JSON array, sorted by condition type.
+func ConditionMarshal(conditions []metav1.Condition) ([]byte, error) {
+	var condCopy []metav1.Condition
+	condCopy = append(condCopy, conditions...)
```

Comment:
what is the diff between this and the above CopyConditions? Maybe I am missing it

Reply:
None. AFAIK. Two different ways to do the same thing, a shallow copy.

Now, I have changed the implementation of the `CopyConditions` function. In the `CopyConditions` function a deep copy is done whereas in the `ConditionMarshal` function a shallow copy is done.

Reply:
Ok now I see, does it make sense to call it DeepCopyConditions?